rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/sqlserver/SqlServerUndoDeleteExecutor.java

```diff
@@ -16,21 +16,32 @@
  */
 package org.apache.seata.rm.datasource.undo.sqlserver;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.stream.Collectors;
-
 import org.apache.seata.common.exception.ShouldNeverHappenException;
 import org.apache.seata.common.util.CollectionUtils;
+import org.apache.seata.common.util.IOUtil;
+import org.apache.seata.rm.datasource.ConnectionProxy;
 import org.apache.seata.rm.datasource.sql.struct.Field;
+import org.apache.seata.rm.datasource.sql.struct.KeyType;
 import org.apache.seata.rm.datasource.sql.struct.Row;
 import org.apache.seata.rm.datasource.sql.struct.TableRecords;
 import org.apache.seata.rm.datasource.undo.SQLUndoLog;
 import org.apache.seata.sqlparser.util.ColumnUtils;
 import org.apache.seata.sqlparser.util.JdbcConstants;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
 
 public class SqlServerUndoDeleteExecutor extends BaseSqlServerUndoExecutor {
+
+    private boolean tableIdentifyExistence = false;
```

Comment:
I hope this variable will eventually be cached in TableMeta and the caching-related refresh and retrieval will be implemented in SqlServerTableMetaCache, rather than in the executor, where it needs to be executed every time executeOn is called.

Reply:
Originally, I  considered putting it in a cache. But then I thought of that the application has to be started again to refresh the cache when a column that has something to do with ""INDENTITY  changed. So I quit to do that. 
Now I rethink that it's probable for a application to restart when a column is changed. 
I will fix it tonight.

Reply:
Done!
I use the SqlServerTableMetaCache to help me. Please help me reivew it again!
![image](https://github.com/apache/incubator-seata/assets/55011193/e4fad5b7-a35d-4373-b2fd-17dc531121d5)

Reply:
My idea is to have `tableIdentifyExistence` handled together with `resultSetMetaToSchema` in `SqlServerTableMetaCache`, rather than handling it separately in `SqlServerUndoDeleteExecutor`.

Reply:
I pondered about it last night. And I found that the cahce key should be a combination of resourceId, schema and table name. It may be a feasible way to put it in the subclass of TableMeta. But when I was creating a subclass of TableMeta named SqlServerTableMeta, I found it a little elegant again. So I make a compromise and fix it in this way. What do you think of it? Any better ideas？

Reply:
There is a typo there. It should be " I found it a little not elegant again". o(╯□╰)o

Reply:
I found it a good way to judge whether  there is a "IDENTITY" in a table by creating a subclass of TableMeta named SqlServerTableMeta with a new filed in it for justification. But I found there are no other subclass of TableMeta for other SQL dialect. Maybe it sounds a bit inelegant, too. What do you think of it? @funky-eyes

Reply:
> I found it a good way to judge whether there is a "IDENTITY" in a table by creating a subclass of TableMeta named SqlServerTableMeta with a new filed in it for justification. But I found there are no other subclass of TableMeta for other SQL dialect. Maybe it sounds a bit inelegant, too. What do you think of it? @funky-eyes

I think it's possible to create a subclass to provide specific metadata for different databases. I agree with your approach, let's go ahead and do it.

Reply:
Done! Please review! Thanks!

Reply:
@funky-eyes Please help me review! Thanks!